omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -421,58 +429,86 @@ public static void setValue(Obs obs, Object value) throws ParseException, Conver
 					} else {
 						throw new ObjectNotFoundException(obs.getConcept().getName().getName() + ":" + value.toString());
 					}
-					
+
 				} else {
 					obs.setValueCoded(valueCoded);
 				}
-				
-			} else {
-				if (obs.getConcept().isNumeric()) {
-					//get the actual persistent object rather than the hibernate proxy
-					ConceptNumeric concept = Context.getConceptService().getConceptNumeric(obs.getConcept().getId());
-					String units = concept.getUnits();
-					if (StringUtils.isNotBlank(units)) {
-						String originalValue = value.toString().trim();
-						if (originalValue.endsWith(units))
-							value = originalValue.substring(0, originalValue.indexOf(units)).trim();
-						else {
-							//check that that this value has no invalid units
-							try {
-								Double.parseDouble(originalValue);
-							}
-							catch (NumberFormatException e) {
-								throw new APIException(originalValue + " has invalid units", e);
-							}
+				return;
+			}
+
+			// special case for Location
+			String potentialLocationUuid = null;
+
+			// if this is a representation of an object, so get uuid property as potential location uuid
+			if (value instanceof Map) {
+				Object uuid = ((Map) value).get(RestConstants.PROPERTY_UUID);
+				if (uuid != null) {
+					potentialLocationUuid = uuid.toString();
+				}
+			}
+			else {
+				// otherwise, could this just uuid value
+				potentialLocationUuid = value.toString();
+			}
+
+			// if there is a potential uuid, see if there is a matching location and set
+			if (RestUtil.isUuid(potentialLocationUuid)) {
```

Comment:
We don't strictly need to do a uuid check, but otherwise we'd we calling a getLocation.getLocationByUuid for any value text passed it, which seemed like it could be a performance hit.